src/main/java/org/jabref/logic/search/indexing/BibFieldsIndexer.java

```diff
@@ -239,7 +242,12 @@ private void addToIndex(BibEntry bibEntry) {
                 }
                 // endregion
             }
-
+            // ensure all date-related fields are indexed.
+            for (Field dateField : dateFields) {
+                Optional<String> resolvedDateValue = bibEntry.getResolvedFieldOrAlias(dateField, this.databaseContext.getDatabase());
+                String dateValue = resolvedDateValue.orElse("");
+                addBatch(preparedStatement, entryId, dateField, dateValue);
```

Comment:
I know, Optionals are hard to learn, please:

Use the methods of java.util.Optional. `ifPresent`.

NOT

```java
Optional<String> resolved = bibEntry.getResolvedFieldOrAlias(...);
String value = resolved.orElse("");
doSomething(value)
```

BUT

```java
Optional<String> resolved = bibEntry.getResolvedFieldOrAlias(...);
resolved.ifPresent(value -> doSomething(value));
```